preview-env/comment/action.yml

```diff
@@ -0,0 +1,51 @@
+---
+name: Manage Deployment Result Summary
+description: |
+  Manages a deployment result summary as comment in the current PR based on deployment artifacts.
+  If deployment artifacts have been uploaded, the action upserts the summary.
+  If there are no deployment artifacts, the summary will get deleted.
+  The deployment artifact name pattern: `deployment-status-<runID>-<runAttempt>-*`
+runs:
+  using: composite
+  steps:
+    - name: Download Deployment Status Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: deployment_status
+        pattern: deployment-status-${{ github.run_id }}-${{ github.run_attempt }}-*
+    - name: Check for deployment artifacts
+      id: check_artifacts
+      shell: bash
+      run: |
+        if [ -d deployment_status ] && find deployment_status -type f -name '*.md' | grep -q .; then
+          echo "ARTIFACTS_FOUND=true" >> $GITHUB_OUTPUT
+        else
+          echo "ARTIFACTS_FOUND=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Create deployment status summary
+      if: steps.check_artifacts.outputs.ARTIFACTS_FOUND == 'true'
+      shell: bash
+      run: |
+        shopt -s globstar   # enables the `**` wildcard
+        summary="## :rocket: Deployment Results :rocket:\n"
+        summary+="Please find below the results of the latest deployments.\n"
+        summary+=$(cat deployment_status/**/*.md)
+        summary+="\n## :wrench: Troubleshooting :wrench:\n"
+        summary+="The :link: \`ArgoCD\` link can be used to check the state and configuration of all the services deployed as part of the preview environments and get logs.\n\n"
+        summary+="In case of error, please check :clipboard: \`Deployment Jobs\` and :link: \`ArgoCD\` to debug, and check our (yet to come) troubleshooting page"
```

Comment:
Could we use a template file(s) here (as you did for the `create` composite action) ? It would be better for maintenance and readability.

Reply:
I just saw the following for the `message-path` attribute: https://github.com/mshick/add-pr-comment?tab=readme-ov-file#configuration-options

> Supports multi-line input and globs. Multiple messages will be concatenated.

Meaning you can eventually simplify all this with something like

```yaml
message-path: |
  templates/head.md
  deployment_status/**/*.md
  templates/troubleshooting.md
```